src/FieldWarning/Assets/Camera/SlidingCameraBehaviour.cs

```diff
@@ -97,9 +99,38 @@ public class SlidingCameraBehaviour : MonoBehaviour
 
     private Terrain _terrain;
 
+    [Serializable]
+    private struct TerrainMaterial
+    {
+#pragma warning disable 0649
+        public Material Material;
+        public float MaxAltitude;
+#pragma warning restore 0649
+    }
+
+    [SerializeField]
+    [Tooltip("Must be sorted in ascending order. The first material whose max altitude "
+        + "is higher than the current camera altitude will be applied to the terrain.")]
+    List<TerrainMaterial> _terrainMaterials = null;
+    [SerializeField]
+    private MicroSplatTerrain _microSplatTerrain = null;
+
     private void Awake()
     {
         _terrain = Terrain.activeTerrain;
+
+        if (_microSplatTerrain == null) {
+            _microSplatTerrain = _terrain.GetComponent<MicroSplatTerrain>();
+        }
+
+        if (_microSplatTerrain == null) {
+            throw new Exception("Camera not set up correctly, microsplat reference missing!");
+        }
+
+        if (_terrainMaterials == null || _terrainMaterials.Count == 0) {
+            throw new Exception("Camera not set up correctly, terrain materials missing!");
+        }
+
```

Comment:
In a lot of places we're omitting the braces for single-line conditionals. I think we should pick one convention (always include or omit when possible) and stick with it, and maybe add it to the coding standards doc.

I'm personally a fan of omitting them, but there are good arguments for always including them. Thoughts?

Reply:
I can live with any of the three possibilities as long as I don't have to adapt the codebase myself ^^

Reply:
lol no worries, I'll look into it at some point